File: Cs/Solution/FindPermutationDifference.cs
using System;

namespace Solution
{
	public partial class Solution
	{
		public int FindPermutationDifference(string s, string t)
		{
			int[] cnt = new int[26];
			for (int i = 0; i < s.Length; i++)
			{
				cnt[s[i] - 'a'] = i;
			}
			int ans = 0;
			for (int i = 0; i < t.Length; i++)
			{
				ans += Math.Abs(cnt[t[i] - 'a'] - i);
			}
			return ans;
		}
	}
}

File: Rust/src/solution/find_permutation_difference.rs
use super::Solution;

impl Solution {
    pub fn find_permutation_difference(s: String, t: String) -> i32 {
        let mut map = vec![0; 26];
        let mut sum = 0;
        for i in 0..s.len() {
            map[(s.as_bytes()[i] - b'a') as usize] = i;
        }
        for i in 0..t.len() {
            let idx = (t.as_bytes()[i] - b'a') as usize;
            sum += (i as i32 - map[idx] as i32).abs();
        }
        sum
    }
}
